org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/CompilerBridgeStore.scala

```diff
@@ -37,8 +41,20 @@ class CompilerBridgeStore(base: IPath, plugin: ScalaPlugin) extends HasLogger {
   // raw stats
   private var hits, misses = 0
 
-  private lazy val compilerBridgeSrc =
-    OSGiUtils.getBundlePath(plugin.zincCompilerBridgeBundle).flatMap(EclipseUtils.computeSourcePath(SdtConstants.ZincCompilerBridgePluginId, _))
+  private lazy val compilerBridgeSrc = new Function[ScalaVersion, Option[IPath]] {
+    private[this] lazy val defaultPath = OSGiUtils.getBundlePath(plugin.zincCompilerBridgeBundle).flatMap(EclipseUtils.computeSourcePath(SdtConstants.ZincCompilerBridgePluginId, _))
+    @volatile private[this] var path_2_10: Option[IPath] = None
+    override def apply(scalaVersion: ScalaVersion) = synchronized {
+      scalaVersion match {
+        case SpecificScalaVersion(2, 10, _, _) =>
+          if (path_2_10.isEmpty)
+            path_2_10 = OSGiUtils.getBundlePath(plugin.zincCompilerBridgeBundle).flatMap(EclipseUtils.computeSourcePath(SdtConstants.ZincCompilerBridgePluginId, _, scalaVersion))
+          path_2_10
+        case _ =>
+          defaultPath
+      }
+    }
+  }.apply(_)
```

Comment:
This can also be written as
```scala
lazy val compilerBridgeSrc = {
  val defaultPath = ...
  var path_2_10 = ...
  (scalaVersion: ScalaVersion) => ...
}
```